care/facility/api/serializers/patient_consultation.py

```diff
@@ -332,6 +354,55 @@ def create(self, validated_data):
 
         return consultation
 
+    def validate_create_diagnoses(self, value):
+        # Reject if create_diagnoses is present for edits
+        if self.instance and value:
+            raise ValidationError("Bulk create diagnoses is not allowed on update")
+
+        # Reject if no diagnoses are provided
+        if len(value) == 0:
+            raise ValidationError("Atleast one diagnosis is required")
+
+        # Reject if duplicate diagnoses are provided
+        if len(value) != len(set([obj["diagnosis"].id for obj in value])):
+            raise ValidationError(
+                {"create_diagnoses": ["Duplicate diagnoses are not allowed"]}
+            )
+
+        principal_diagnosis, confirmed_diagnoses = None, []
+        for obj in value:
+            if obj["verification_status"] == ConditionVerificationStatus.CONFIRMED:
+                confirmed_diagnoses.append(obj)
+
+            # Reject if there are more than one principal diagnosis
+            if obj["is_principal"]:
+                if principal_diagnosis:
+                    raise ValidationError(
+                        {
+                            "create_diagnoses": [
+                                "Only one diagnosis can be set as principal diagnosis"
+                            ]
+                        }
+                    )
+                principal_diagnosis = obj
+
+        # Reject if principal diagnosis is not one of confirmed diagnosis (if it is present)
+        if (
+            principal_diagnosis
+            and len(confirmed_diagnoses)
```

Comment:
```suggestion
```
redunddant because of `principal_diagnosis["verification_status"] != ConditionVerificationStatus.CONFIRMED`

Reply:
No, What if we have 2 provisional diagnoses with one of them set as principal and 0 confirmed diagnoses? This is still considered valid right?